app/middlewares/lead_provider_request_auditor.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class LeadProviderRequestAuditor
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    store_request_details(env)
+    @app.call(env)
+  end
+
+private
+
+  def store_request_details(env)
+    path = env["PATH_INFO"]
+    if audited_paths.any? { |pattern| pattern.match?(path) }
+      request = Rack::Request.new(env)
+      body = request.body.read.squish
```

Comment:
Do we need to store a hashed token or anything here? How would we use this?

Reply:
I think we would dump the contents of the audited endpoints into some lovely google studio to see something. 

I think until I get some concrete use cases (which so far I don't think we have), I prefer to keep it as simple as possible.